### ISSUE TITLE
[monorepo] ESLint consolidation 

### ISSUE THREAD

**Issue by Alonski** [statechannels/monorepo#652]
In #611 there was a fair bit of discussion surround various ESLint settings and configurations.

To unblock the merging of that PR I am moving the discussion to here.

Things to discuss:
1. @snario  Many of the rules in .eslintignore overlap with (a) .gitignore regexes and (b) other .eslintignore in other packages. Can we consolidate?

2. @snario Many of the rules in package eslintrc.js seem to be duplicates of rules in other packages. Can we consolidate them in root .eslintrc.js file please?

3. @snario I would be careful with eslint --cache option. When testing a branch I noticed it caused issues when deleting built contract artifacts, running the linter, and then rebuilding them, and running it again. The linter would fail despite the contract artifacts existing.

Lets discuss these things here.

**Comment by snario** [comment]
#686

**Comment by snario** [comment]
Did this earlier, we have a root eslint which others can inherit now.
